### PR TITLE
Clean up unused code in kubelet

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

#### What type of PR is this?
Clean up unused code in kubelet
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes # None

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -232,8 +232,7 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 			logs.InitLogs()
 			logOption := &logs.Options{Config: kubeletConfig.Logging}
 			if err := logOption.ValidateAndApply(utilfeature.DefaultFeatureGate); err != nil {
-				klog.ErrorS(err, "Failed to initialize logging")
-				os.Exit(1)
+				return fmt.Errorf("failed to initialize logging: %w", err)
```
Comment (aojea) [inline cmd/kubelet/app/server.go:235]: This was part of https://github.com/kubernetes/kubernetes/commit/4df70dfd41601077478289ad358e7c10a4f0d0e1

What has changed to remove it?

Comment (yxxhero) [reply cmd/kubelet/app/server.go:235]: The logic is the same, except that the upper layer performs log output and exit. @aojea WDTF?

Comment (aojea) [reply cmd/kubelet/app/server.go:235]: WDTF?? or WDTY?? :smile:

**Comment by aojea** [comment]
/assign @pohly 

he is the best person, or at least one of them to review this,

File: cmd/kubelet/app/server.go (hunk 1 above)
Comment (yxxhero) [reply cmd/kubelet/app/server.go:235]: WDTY got it! 😄😄😄😄😄😄

**Comment by pohly** [comment]
/lgtm

```console
$ go run ./cmd/kubelet --logging-format=xyz 
Flag --logging-format has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
E0411 10:06:00.034229   40869 run.go:74] "command failed" err="failed to validate kubelet configuration, error: logging.format: Invalid value: \"xyz\": Unsupported log format, path: &TypeMeta{Kind:,APIVersion:,}"
exit status 1
```

It would be slightly nicer if that error wasn't reported with `klog.Error` because logging hasn't been fully initialized yet. But that's a problem of `cli.Run` having to guess how far the application got before encountering the error, i.e. unrelated to this change.

**Review by cndoit18 — APPROVED** [review]
LGTM

**Review by matthyx — APPROVED** [review]
/lgtm

**Review by matthyx — APPROVED** [review]
/lgtm

**Comment by yxxhero** [comment]
@matthyx Thanks very much

**Review by mrunalp — APPROVED** [review]

**Pull request merged**